### PR TITLE
Force full backup on specified day in daily schedule

### PR DESCRIPTION
**Pull request by ram-infrac**
**What type of PR is this?**
>improvement

**What this PR does / why we need it**:
Daily schedule policy now accept `forceFullSnapshotDay` day which will result in force full backup of volume 

**Does this PR change a user-facing CRD or CLI?**:
yes

```
apiVersion: stork.libopenstorage.org/v1alpha1
kind: VolumeSnapshotSchedule
metadata:
  name: mysql-snapshot-schedule
  namespace: test
  annotations:
    portworx/snapshot-type: cloud
spec:
  schedulePolicyName: daily
  suspend: false
  reclaimPolicy: Delete
  template:
    spec:
      persistentVolumeClaimName: mysql-data
```

Schedule Policy : 
```
apiVersion: stork.libopenstorage.org/v1alpha1
kind: SchedulePolicy
metadata:
  name: daily
policy:
  daily:
    time: "12:04PM"
    forceFullSnapshotDay: "Monday"
```

**Is a release note needed?**:
yes

```release-note
User can specify `forceFullSnapshotDay`  in daily snapshot schedule policy to ensure full volume backup on specified day of the week
```

**Does this change need to be cherry-picked to a release branch?**:
yes, 2.12.0

### CODE REVIEW

File: pkg/apis/stork/v1alpha1/schedulepolicy.go [hunk 1]
```diff
@@ -124,6 +124,8 @@ type DailyPolicy struct {
 	// Options to be passed in to the driver. These will be passed in
 	// to the object being triggered
 	Options map[string]string `json:"options"`
+	// ForceFullSnapshotDay on day specified.
```
Comment (lpitstickpstg) [inline pkg/apis/stork/v1alpha1/schedulepolicy.go:127]: ... "specifies day of the week for full snapshot to take place"

File: pkg/schedule/schedule.go [hunk 2]
```diff
@@ -30,7 +30,8 @@ const (
 	// MockTimeConfigMapNamespace is the namespace of the config map used to mock times
 	MockTimeConfigMapNamespace = "kube-system"
 	// MockTimeConfigMapKey is the key name in the config map data that contains the time
-	MockTimeConfigMapKey = "time"
+	MockTimeConfigMapKey       = "time"
+	incrementalCountAnnotation = "portworx.io/cloudsnap-incremental-count"
```
Comment (lpitstickpstg) [inline pkg/schedule/schedule.go:34]: re-definition of constant?

**Review by lpitstickpstg — CHANGES_REQUESTED** [review]
Your example does not contain an example of this being used in spec. We should have a good reference that can also be used in our docs.

Release note should say "... specified day of the week"

**Review by lpitstickpstg — APPROVED** [review]

File: pkg/schedule/schedule.go [hunk 3]
```diff
@@ -229,7 +230,17 @@ func GetOptions(policyName string, namespace string, policyType stork_api.Schedu
...
-		return schedulePolicy.Policy.Daily.Options, nil
+		options := schedulePolicy.Policy.Daily.Options
+		scheduledDay, ok := stork_api.Days[schedulePolicy.Policy.Daily.ForceFullSnapshotDay]
+		if ok {
+			currentDay := GetCurrentTime().Weekday()
+			// force full backup on specified day
+			if currentDay == scheduledDay {
+				options[utils.PXIncrementalCountAnnotation] = "0"
+			}
+			logrus.Debugf("Forcing full-snapshot for daily snapshotschedule policy on the day %s", schedulePolicy.Policy.Daily.ForceFullSnapshotDay)
```
Comment (adityadani) [inline pkg/schedule/schedule.go:241]: This can be an info log line. Will be helpful while debugging.

**Review by adityadani — APPROVED** [review]

**Pull request merged**